src/main/java/ch/njol/skript/lang/SyntaxElement.java

```diff
@@ -47,4 +48,22 @@ default ParserInstance getParser() {
 		return ParserInstance.get();
 	}
 
+	/**
+	 * Gets the structures in which this element can be used in. If there are no limits to this element, you may
+	 * return an empty array, or simply not override this method.
+	 * @return the structures this element is usable in, or an empty array.
+	 */
+	default Class<? extends Structure>[] getUsableStructures() {
+		return new Class[0];
+	}
+
+	/**
+	 * Gets the sections in which this element can be used in. If there are no limits to this element, you may
+	 * return an empty array, or simply not override this method.
+	 * @return the sections this element is usable in, or an empty array.
+	 */
+	default Class<? extends Section>[] getUsableSections() {
+		return new Class[0];
+	}
+
```

Comment:
You probably want this to use lists instead of arrays

Reply:
I was a little unsure about this myself. Do you think list+steams would be better here than array+iteration? Not sure if performance is really a factor here though.

Reply:
I just wanted to keep consistent with the arrays being primarily used, performance with the streams wouldnt be too much different especially if it improves readability.

Reply:
Lists+Streams should be good then IMO

Might be a good idea to have the default implementations return an unmodifiable list too

Reply:
I do think list+streams are better, with streams you tell what the result needs to be instead of what to do and lists allow you to prevent outside modification, the performance impact is close to zero because of JIT optimization.

Reply:
shall change 👍🏼